subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -264,11 +265,14 @@ public void stopWatchingAfterError() {
                     stopWatchingAndInvalidateHierarchyAfterError();
                 }
             });
-            watchableHierarchies.forEach(watchableHierarchy -> watchRegistry.registerWatchableHierarchy(watchableHierarchy, currentRoot));
+            SnapshotHierarchy newRoot = watchRegistry.buildStarted(currentRoot.empty(), watchMode);
```

Comment:
I think we should rename these methods `updateVfsOnBuildStarted()` and `updateVfsOnBuildFinished()`. WDYT?

Reply:
I renamed those methods to `updateVfsOnBuild...`.